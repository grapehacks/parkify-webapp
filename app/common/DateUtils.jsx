class DateUtils {

    static getDayString(date) {
        var weekdays = this.getWeekDaysNames();
        return weekdays[date.getDay()];
    }

    static getMonthDayString(date) {
        var dateString = date.getDate();
        var monthString = this.getMonth(date);
        return dateString + ' ' + monthString;
    }

    static getMonth(date) {
        var months = this.getMonthNames();
        return months[date.getMonth()];
    }

    static getMonthNames() {
        return ['January', 'February', 'March', 'April', 'May', 'June',
            'July', 'August', 'September', 'October', 'November', 'December'
        ];
    }

    static getWeekDaysNames() {
        return [ 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
    }

}

module.exports = DateUtils;