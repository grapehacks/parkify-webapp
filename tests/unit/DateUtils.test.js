import DateUtils from '../../app/common/DateUtils';

describe('DateUtils', () => {

    it('should return correct month names', () => {
        expect(DateUtils.getMonthNames()).toMatchSnapshot();
    });

    it('should return correct week day names', () => {
        expect(DateUtils.getWeekDaysNames()).toMatchSnapshot();
    });

    it('should return correct month name from date object', () => {
        var d = new Date();
        d.setFullYear(2017, 3, 15); // april
        expect(DateUtils.getMonth(d)).toMatchSnapshot();
    });

    it('should return correct \'month and day\' string', () => {
        var d = new Date();
        d.setFullYear(2017, 3, 15); // april
        // 15 April
        expect(DateUtils.getMonthDayString(d)).toMatchSnapshot();
    });

    it('should return correct day name from date object', () => {
        var d = new Date();
        d.setFullYear(2017, 3, 15); // april
        // Saturday
        expect(DateUtils.getDayString(d)).toMatchSnapshot();
    });
});