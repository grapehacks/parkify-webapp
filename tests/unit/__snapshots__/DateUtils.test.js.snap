// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateUtils should return correct 'month and day' string 1`] = `"15 April"`;

exports[`DateUtils should return correct day name from date object 1`] = `"Saturday"`;

exports[`DateUtils should return correct month name from date object 1`] = `"April"`;

exports[`DateUtils should return correct month names 1`] = `
Array [
  "January",
  "February",
  "March",
  "April",
  "May",
  "June",
  "July",
  "August",
  "September",
  "October",
  "November",
  "December",
]
`;

exports[`DateUtils should return correct week day names 1`] = `
Array [
  "Sunday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
]
`;
